src/values.rs

```diff
@@ -257,8 +311,52 @@ impl PartialOrd for ExtNumeric {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct Binary {
+   /// The actual data. We put it behind an `Arc` to make sure
+   /// that cloning remains inexpensive.
+   data: Arc<Vec<u8>>,
+   /// The mime type. Should probably be an Id<MimeTypeId>.
+   mimetype: String
+}
+
 /// Representation of an actual value that can be sent to/received
 /// from a service.
+///
+/// # (De)serialization
+///
+/// Values of this state are represented by an object `{ key: value }`, where key is one of
+/// `Unit`, `OnOff`, `OpenClosed`, ... Note that the `value` for `Unit` is `[]`.
+///
+/// ```
+/// extern crate serde;
+/// extern crate serde_json;
+/// extern crate foxbox_taxonomy;
+///
+/// # fn main() {
+/// use foxbox_taxonomy::values::Value::*;
+/// use foxbox_taxonomy::values::OnOff::*;
+/// use foxbox_taxonomy::values::OpenClosed::*;
+///
+/// let unit = serde_json::to_string(&Unit).unwrap();
+/// assert_eq!(unit, "{\"Unit\":[]}");
+///
+/// let unit : foxbox_taxonomy::values::Value = serde_json::from_str("{\"Unit\":[]}").unwrap();
+/// assert_eq!(unit, Unit);
+///
+/// let on = serde_json::to_string(&OnOff(On)).unwrap();
+/// assert_eq!(on, "{\"OnOff\":\"On\"}");
+///
+/// let on : foxbox_taxonomy::values::Value = serde_json::from_str("{\"OnOff\":\"On\"}").unwrap();
+/// assert_eq!(on, OnOff(On));
+///
+/// let open = serde_json::to_string(&OpenClosed(Open)).unwrap();
+/// assert_eq!(open, "{\"OpenClosed\":\"Open\"}");
+///
+/// let open : foxbox_taxonomy::values::Value = serde_json::from_str("{\"OpenClosed\":\"Open\"}").unwrap();
```

Comment:
__question:__ do we have any agreement on max line length? I've seen ```99``` in [1], but not sure if it applies to comments as well.

[1] https://github.com/rust-lang/rust/blob/master/src/doc/style/style/whitespace.md